opencensus/trace/span_context.py

```diff
@@ -149,13 +146,13 @@ def _check_trace_id(self, trace_id):
 
 
 def generate_span_id():
-    """Return the random generated span ID for a span. Must be a 16 character
-    hexadecimal encoded string
+    """Return the random generated span ID for a span. Must be a 64 bit
```

Comment:
I have a different understanding.
`SpanContext` is the in-memory representation, where we prefer span_id as byte array for sake of performance and interoperability with different protocols. https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/Span.md#spanid

The actual propagator will be in charge of converting the incoming request headers into the in memory representation, or converting the in-memory representation to outgoing request headers. Examples are https://github.com/census-instrumentation/opencensus-python/blob/master/opencensus/trace/propagation/b3_format.py#L61 and https://github.com/census-instrumentation/opencensus-python/blob/master/opencensus/trace/propagation/trace_context_http_header_format.py#L55.

In order to have the correct implementation for Stackdriver format, what you need to change is the serialization and deserialization logic inside `google_cloud_format.py`, rather than the `SpanContext` data structure.

Reply:
I understand, out of curiosity, what is the advantage of having it as hexadecimal representation internally? Is it because most other representations use hex, and Stackdriver is the exception?

Reply:
I believe it was chosen due to the performance consideration + being neutral to the actual protocols.

You can refer to https://github.com/census-instrumentation/opencensus-proto/blob/master/src/opencensus/proto/trace/v1/trace.proto#L60.